apphub/semantic_segmentation/unet3d/3DUnetMRBrainSegmentation_Isensee.py

```diff
@@ -0,0 +1,197 @@
+        cond = tf.logical_or(cond1, cond2)
+        data_patch, truth_patch = tf.cond(cond,lambda: self.handle_padding_case(patch),
+                                          lambda:self.handle_nopaddding_case(patch))
+        return data_patch, truth_patch
+
+    def forward(self, data, state):
+        self.modality_images, self.seg_mask = data
+        cls_1 = tf.where(self.seg_mask[0] == self.labels[0], 1, 0)
+        cls_2 = tf.where(self.seg_mask[0] == self.labels[1], 1, 0)
+        cls_4 = tf.where(self.seg_mask[0] == self.labels[2], 1, 0)
+        self.segmask_multi_class = tf.stack([cls_1, cls_2, cls_4])
+
+        # We randomly select the start of the patches
+        random_start_offset = tf.negative(
+            tf.random.uniform(shape=(3, ), minval=0, maxval=self.patch_start_offset + 1, dtype=tf.dtypes.int32))
+
+        start = random_start_offset
+        stop = self.image_shape + random_start_offset
+        step = self.patch_shape
+        x,y,z = tf.meshgrid( tf.range(start[0],stop[0],step[0]),  tf.range(start[1],stop[1],step[1]),
+                                        tf.range(start[2],stop[2],step[2]),indexing='ij')
+        patches = tf.stack([x, y, z], axis=0)
+        patches = tf.transpose(tf.reshape(patches, (3, -1)))
+
+        data_stack,truth_stack = tf.map_fn(self.get_image_patch,patches,
+                                           dtype=(tf.dtypes.float32, tf.dtypes.int32), back_prop=False)
+        mask = tf.map_fn(lambda tr: tf.reduce_any(tf.not_equal(tr, 0)), truth_stack, dtype=tf.bool, back_prop=False)
+        data_stack = tf.boolean_mask(data_stack, mask)
+        truth_stack = tf.boolean_mask(truth_stack, mask)
+
+        return data_stack, truth_stack
+
+
+class SplitMaskLabelwise(TensorOp):
+    def __init__(self, inputs=None, outputs=None, mode=None):
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode)
+        self.nlabels = 3
+        self.labels = [1, 2, 4]
+
+    def forward(self, data, state):
+        seg_mask = data
+        cls_1 = tf.where(seg_mask[0] == self.labels[0], 1, 0)
+        cls_2 = tf.where(seg_mask[0] == self.labels[1], 1, 0)
+        cls_4 = tf.where(seg_mask[0] == self.labels[2], 1, 0)
+        segmask_multi_label = tf.stack([cls_1, cls_2, cls_4])
+        return segmask_multi_label
+
+
+
+def get_estimator(batch_size=1, epochs=500, steps_per_epoch=128, model_dir=tempfile.mkdtemp(), path_brats=None):
+    """Args:
+        path_brats: folder path of BraTs 2018 dataset, containing data subdir inturn having LGG and HGG data.
+        Expected folder structure is
+        /home/ubuntu/Brats
+        /home/ubuntu/Brats/data
+        /home/ubuntu/Brats/data/LGG
+        /home/ubuntu/Brats/data/HGG
+    """
+    assert path_brats is not None, 'Pass valid folder path of BraTs 2018 dataset'
+    # Ensure Brats 2018 dataset is downloaded. Pass the folder contianing train and val subdirectories.
+    # currently the script doesn't download the ImageNet data.
```

Comment:
why would the script download ImageNet data?

Reply:
Typo. Cleaned it.